File: frontend/src/types/satellite.ts
export type Satellite = {
  satellite_id: number
  name: string
  norad_id: number
  object_id: string | null
  country: string | null
  mission_type: string
  orbit_type: string | null
  inclination_deg: number | null
  orbital_period_min: number | null
  avg_altitude_km: number | null
  description: string | null
  data_access_type: string | null
  data_access_note: string | null
}

export type SensorBand = {
  band_id: number
  sensor_id: number
  // band_name: string
  spectral_range_nm: string | null
  spatial_resolution_m: number
  band_code: string | null
  band_name: string | null
  wavelength_min_nm: number | null
  wavelength_max_nm: number | null
  band_type: string | null
  is_grouped: boolean
}

export type Sensor = {
  sensor_id: number
  satellite_id: number
  name: string
  sensor_type: string
  swath_km: number | null
  off_nadir_max_deg: number | null
  retarget_time_sec: number | null
  notes: string | null
  bands: SensorBand[]
  data_access_type: string | null
  data_access_note: string | null
  max_off_nadir_deg: number | null
  modes: SensorMode[]
}

export type SensorMode = {
  sensor_mode_id: number
  sensor_id: number
  mode_name: string
  mode_type: string | null
  swath_km: number | null
  spatial_resolution_m: number | null
  max_off_nadir_deg: number | null
  is_default: boolean
  description: string | null
}

File: backend/app/schemas/satellite.py
from pydantic import BaseModel, ConfigDict


class SensorBandRead(BaseModel):
    band_id: int
    sensor_id: int
    band_code: str | None = None
    band_name: str | None = None
    spectral_range_nm: str | None = None
    wavelength_min_nm: float | None = None
    wavelength_max_nm: float | None = None
    spatial_resolution_m: float | None = None
    band_type: str | None = None
    is_grouped: bool = False

    model_config = ConfigDict(from_attributes=True)


class SensorModeRead(BaseModel):
    sensor_mode_id: int
    sensor_id: int
    mode_name: str
    mode_type: str | None = None
    swath_km: float | None = None
    spatial_resolution_m: float | None = None
    max_off_nadir_deg: float | None = None
    is_default: bool = False
    description: str | None = None

    model_config = ConfigDict(from_attributes=True)


class SensorRead(BaseModel):
    sensor_id: int
    satellite_id: int
    name: str
    sensor_type: str | None = None
    swath_km: float | None = None
    spatial_resolution_m: float | None = None
    notes: str | None = None
    bands: list[SensorBandRead] = []
    modes: list[SensorModeRead] = []

    model_config = ConfigDict(from_attributes=True)


class SatelliteRead(BaseModel):
    satellite_id: int
    name: str
    norad_id: int
    object_id: str | None = None
    country: str | None = None
    mission_type: str | None = None
    orbit_type: str | None = None
    inclination_deg: float | None = None
    orbital_period_min: float | None = None
    avg_altitude_km: float | None = None
    description: str | None = None
    data_access_type: str | None = None
    data_access_note: str | None = None

    model_config = ConfigDict(from_attributes=True)